src/object/object_primitive.c

```diff
@@ -7830,50 +7829,41 @@ pr_midxkey_compare (DB_MIDXKEY * mul1, DB_MIDXKEY * mul2, int do_coercion, int t
   }
 #endif /* NDEBUG */
 
-  size1 = size2 = 0;
-
-  mem1 = bitptr1 = mul1->buf;
-  mem2 = bitptr2 = mul2->buf;
+  bitptr1 = mul1->buf;
+  bitptr2 = mul2->buf;
 
-  adv_size1 = OR_MULTI_BOUND_BIT_BYTES (mul1->domain->precision);
+  adv_size = OR_MULTI_BOUND_BIT_BYTES (mul1->domain->precision);
 
-  mem1 += adv_size1;
-  mem2 += adv_size1;
-  size1 += adv_size1;
-  size2 += adv_size1;
+  mem1 = bitptr1 + adv_size;
+  mem2 = bitptr2 + adv_size;
 
   dom1 = mul1->domain->setdomain;
   dom2 = mul2->domain->setdomain;
 
-  if (num_index_term > 0)
-    {
-      last = num_index_term;
-    }
-  else
-    {
-      last = mul1->ncolumns;
-    }
+  last = (num_index_term > 0) ? num_index_term : mul1->ncolumns;
 
-  for (i = 0; start_colp && i < *start_colp; i++, dom1 = dom1->next, dom2 = dom2->next)
+  i = 0;
+  if (start_colp)
     {
-      if (dom1 == NULL || dom2 == NULL || dom1->is_desc != dom2->is_desc)
+      for ( /* blank */ ; i < *start_colp; i++, dom1 = dom1->next, dom2 = dom2->next)
 	{
-	  assert (false);
-	  return DB_UNK;
-	}
+	  if (dom1 == NULL || dom2 == NULL || dom1->is_desc != dom2->is_desc)
+	    {
+	      assert (false);
+	      return DB_UNK;
+	    }
 
-      if (OR_MULTI_ATT_IS_BOUND (bitptr1, i))
-	{
-	  adv_size1 = pr_midxkey_element_disk_size (mem1, dom1);
-	  mem1 += adv_size1;
-	  size1 += adv_size1;
-	}
+	  if (OR_MULTI_ATT_IS_BOUND (bitptr1, i))
+	    {
+	      adv_size = pr_midxkey_element_disk_size (mem1, dom1);
+	      mem1 += adv_size;
```

Comment:
How about below?
```suggestion
	      mem1 += pr_midxkey_element_disk_size (mem1, dom1);
```

Reply:
I thought Optimizer would take care of that.
I thought it would be nice to have a variable to put the value in for debugging in case.

Reply:
I agree. Thank you.